### DISCUSSION TITLE
How to setup venv on an arm64 raspberry pi

### DISCUSSION THREAD

**Discussion by paulc1983** [pythonnet/pythonnet#2287] [Q&A]
Hi 

I am trying to setup a raspberry pi 64 bit environment with dotnet7.0. I cant use pip to install modules so have setup a virtual environment, but get errors intitializing it: 

`	Loaded '__Python_Runtime_Generated_Assembly0'. 
	  Python path configuration:
	  PYTHONHOME = '/home/paul/Dotnet.Bledom.Api/PythonEnv'
	  PYTHONPATH = (not set)
	  program name = 'python3'
	  isolated = 0
	  environment = 1
	  user site = 1
	  safe_path = 0
	  import site = 1
	  is in build tree = 0
	  stdlib dir = ''
	  sys._base_executable = '/usr/bin/python3'
	  sys.base_prefix = ''
	  sys.base_exec_prefix = ''
	  sys.platlibdir = 'lib'
	  sys.executable = '/usr/bin/python3'
	  sys.prefix = ''
	  sys.exec_prefix = ''
	  sys.path = [
		'/usr/lib/python3/dist-packages/thonny/plugins/cpython_backend;/usr/lib/python311.zip;/usr/lib/python3.11;/usr/lib/python3.11/lib-dynload;/home/paul/Dotnet.Bledom.Api/PythonEnv/lib/python3.11/site-packages;/home/paul/PythonEnv/Dev/lib/python3.11/site-packages;/home/paul/PythonEnv/Dev/local/lib/python3.11/dist-packages;/home/paul/PythonEnv/Dev/lib/python3/dist-packages;/home/paul/PythonEnv/Dev/lib/python3.11/dist-packages',
	  ]
	Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
	Python runtime state: core initialized
	ModuleNotFoundError: No module named 'encodings'`

I have created a stack overflow question with more details and sample code [here](https://stackoverflow.com/questions/77558271/setting-up-pythonnet-to-run-against-a-virtual-environment-on-a-64-bit-raspberryp)

can anyone confirm that this should work on arm64 debian 12 bookworm and python3.11? On a side note does anyone know how to get round the restrictions on pip on a 64 bit externally managed python environment?

Thanks

Paul

**Comment by iFreilicht** [comment]
You're using the wrong PATH separator (`;` instead of `:`), see https://github.com/pythonnet/pythonnet/discussions/2217#discussioncomment-6789448 and https://docs.python.org/3/library/os.html#os.pathsep
